File: python/flatten_test.py
# coding: utf-8

from python import arrays


def test_simple_array():
    assert arrays.flatten([1, 2, 3, 4]) == [1, 2, 3, 4]


def test_two_level_array():
    assert arrays.flatten([[1], [2], [3], [4]]) == [1, 2, 3, 4]


def test_three_level_array():
    assert arrays.flatten([[1, 2, [3]], 4]) == [1, 2, 3, 4]

File: javascript/src/flatten.test.js
const flatten = require('./arrays');

test('[1,2,3,4] -> [1,2,3,4]', () => {
  array = [1,2,3,4]
  expect(flatten(array)).toEqual([1,2,3,4]);
});

test('[[1],[2],[3],[4] -> [1,2,3,4]', () => {
  array = [[1],[2],[3],[4]];
  expect(flatten(array)).toEqual([1,2,3,4]);
});

test('[[1,2,[3]],4] -> [1,2,3,4]', () => {
  array = [[1,2,[3]],4];
  expect(flatten(array)).toEqual([1,2,3,4]);
});